packages/contracts/contracts/voting/majority/MajorityVotingBase.sol

```diff
@@ -269,41 +293,42 @@ abstract contract MajorityVotingBase is
     }
 
     /// @inheritdoc IMajorityVoting
-    function support(uint256 _proposalId) public view virtual returns (uint256) {
+    function isSupportThresholdReached(uint256 _proposalId) public view virtual returns (bool) {
         Proposal storage proposal_ = proposals[_proposalId];
 
-        return _calculatePct(proposal_.tally.yes, proposal_.tally.yes + proposal_.tally.no);
+        return
+            (RATIO_BASE - proposal_.parameters.supportThreshold) * proposal_.tally.yes >
+            proposal_.parameters.supportThreshold * proposal_.tally.no;
```

Comment:
I'm not entirely sure about this calculation. Shouldn't we compare `yes` against `total * supportThreshold`?

We are multiplying `yes * (1 - supportThreshold)` which I don't understand, and comparing it against `threshold * no`

Reply:
I wrote this in the PR description https://github.com/aragon/core/pull/198#issue-1537873218:

 #### The Support Criterion

 For a proposal to pass, the required ratio of yes and no votes must be met:
 $$(1- \texttt{supportThreshold}) \cdot N_\text{yes} > \texttt{supportThreshold} \cdot N_\text{no}.$$

As explained [in our discord](https://discord.com/channels/672466989217873929/1064843552376508438/1065287740981248070), it is obtained as follows:
The support condition for the proposal to pass is
`y / (y+n) > s`
with
`y`: yes votes
`n`: no votes
`s`: support threshold
Starting from the condition, we do
```
    y / (y+n) > s        |*(y+n)
<=>         y > s*(y+n)  | expand
<=>         y > s*y+ s*n |-s*y
<=>   y - s*y > s*n      | factorize
<=>   y*(1-s) > s*n      |
```
and obtain the condition as a term without division. Since, `RATIO_BASE` is on both sides of the inequality as a pre-factor, it cancels out.

Reply:
Maybe it helps to look at the support criterion with $\texttt{supportThreshold} = \frac{1}{2}=50\%$.
In this case
$$(1- \texttt{supportThreshold}) \cdot N_\text{yes} > \texttt{supportThreshold} \cdot N_\text{no}$$
becomes
$$N_\text{yes} > N_\text{no}$$
as you can remove $\texttt{supportThreshold}$ from both sides of the inequality.

Reply:
Makes total sense @Michael-A-Heuer, thank you!
The inequestions looked unintuitive to me, but the expansion is very clear and with the example too. Thanks for the clarification :+1: